eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -354,12 +401,13 @@ public synchronized BackupRegistry get() {
                     1, clientConfig.getCacheRefreshExecutorThreadPoolSize(), 0, TimeUnit.SECONDS,
                     new SynchronousQueue<Runnable>());  // use direct handoff
 
-            scheduleServerEndpointTask(zone);
-
             fetchRegistryGeneration = new AtomicLong(0);
 
             clientAccept = EurekaAccept.fromString(clientConfig.getClientDataAccept());
 
+            eurekaTransport = new EurekaTransport();
+            scheduleServerEndpointTask(eurekaTransport, zone);
```

Comment:
As most clients will not use this feature, lets create this on demand. Ideally, if turned off we should release resources as well. What do you think?